packages/core/upload/admin/src/components/MediaLibraryInput/Carousel/CarouselAssets.js

```diff
@@ -24,65 +25,89 @@ export const CarouselAssets = ({
-            onDeleteAsset={disabled ? undefined : onDeleteAsset}
-            onDeleteAssetFromMediaLibrary={onDeleteAssetFromMediaLibrary}
-            onAddAsset={disabled ? undefined : onAddAsset}
-            onEditAsset={onEditAsset}
-          />
-        ) : (
-          undefined
-        )
-      }
-    >
-      {assets.length === 0 ? (
-        <CarouselSlide
-          label={formatMessage(
-            { id: getTrad('mediaLibraryInput.slideCount'), defaultMessage: '{n} of {m} slides' },
-            { n: 1, m: 1 }
-          )}
-        >
-          <EmptyStateAsset disabled={disabled} onClick={onAddAsset} onDropAsset={onDropAsset} />
-        </CarouselSlide>
-      ) : (
-        assets.map((asset, index) => (
+    <>
+      <Carousel
+        label={label}
+        selectedSlide={selectedAssetIndex}
+        previousLabel={formatMessage({
+          id: getTrad('mediaLibraryInput.actions.previousSlide'),
+          defaultMessage: 'Previous slide',
+        })}
+        nextLabel={formatMessage({
+          id: getTrad('mediaLibraryInput.actions.nextSlide'),
+          defaultMessage: 'Next slide',
+        })}
+        onNext={onNext}
+        onPrevious={onPrevious}
+        hint={hint}
+        error={error}
+        actions={
+          currentAsset ? (
+            <CarouselAssetActions
+              asset={currentAsset}
+              onDeleteAsset={disabled ? undefined : onDeleteAsset}
+              onDeleteAssetFromMediaLibrary={onDeleteAssetFromMediaLibrary}
+              onAddAsset={disabled ? undefined : onAddAsset}
+              onEditAsset={onEditAsset ? () => setIsEditingAsset(true) : undefined}
+            />
+          ) : (
+            undefined
+          )
+        }
+      >
+        {assets.length === 0 ? (
           <CarouselSlide
-            key={asset.id}
             label={formatMessage(
-              {
-                id: getTrad('mediaLibraryInput.slideCount'),
-                defaultMessage: '{n} of {m} slides',
-              },
-              { n: index + 1, m: assets.length }
+              { id: getTrad('mediaLibraryInput.slideCount'), defaultMessage: '{n} of {m} slides' },
+              { n: 1, m: 1 }
             )}
           >
-            <CarouselAsset asset={asset} />
+            <EmptyStateAsset disabled={disabled} onClick={onAddAsset} onDropAsset={onDropAsset} />
           </CarouselSlide>
-        ))
+        ) : (
+          assets.map((asset, index) => (
+            <CarouselSlide
+              key={asset.id}
+              label={formatMessage(
+                {
+                  id: getTrad('mediaLibraryInput.slideCount'),
+                  defaultMessage: '{n} of {m} slides',
+                },
+                { n: index + 1, m: assets.length }
+              )}
+            >
+              <CarouselAsset asset={asset} />
+            </CarouselSlide>
+          ))
+        )}
+      </Carousel>
+
+      {isEditingAsset && (
+        <EditAssetDialog
```

Comment:
ℹ️ Shifting the dialog outside the carousel fixes the keyboard bug when being on an input